modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItTablesApiTest.java

```diff
@@ -104,97 +114,308 @@
+    }
+
+    /**
+     * Trys to create an index which is already created.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testAddIndex() throws Exception {
+        clusterNodes.forEach(ign -> assertNull(ign.tables().table(TABLE_NAME)));
+
+        Ignite ignite0 = clusterNodes.get(0);
+
+        createTable(ignite0, SCHEMA, SHORT_TABLE_NAME);
+
+        addIndex(ignite0, SCHEMA, SHORT_TABLE_NAME);
+
+        assertThrows(IndexAlreadyExistsException.class,
+                () -> addIndex(ignite0, SCHEMA, SHORT_TABLE_NAME));
+
+        addIndexIfNotExists(ignite0, SCHEMA, SHORT_TABLE_NAME);
+    }
+    
+    /**
+     * Trys to create an index which is already created from lagged node.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    @Disabled("IGNITE-15891 Configuration use local state cache internally, but have to look at consensus")
+    public void testAddIndexFromLaggedNode() throws Exception {
+        clusterNodes.forEach(ign -> assertNull(ign.tables().table(TABLE_NAME)));
+        
+        Ignite ignite0 = clusterNodes.get(0);
+        
+        createTable(ignite0, SCHEMA, SHORT_TABLE_NAME);
+    
+        Ignite ignite1 = clusterNodes.get(1);
+    
+        WatchListenerInhibitor ignite1Inhibitor = metastorageEventsInhibitor(ignite1);
+    
+        ignite1Inhibitor.startInhibit();
+    
+        addIndex(ignite0, SCHEMA, SHORT_TABLE_NAME);
+        
+        CompletableFuture addIndesFut = CompletableFuture.runAsync(() -> addIndex(ignite1, SCHEMA, SHORT_TABLE_NAME));
+        CompletableFuture addIndesIfNotExistsFut = CompletableFuture.runAsync(() -> addIndexIfNotExists(ignite1, SCHEMA, SHORT_TABLE_NAME));
+    
+        for (Ignite ignite : clusterNodes) {
+            if (ignite != ignite1) {
+                assertThrows(IndexAlreadyExistsException.class,
+                        () -> addIndex(ignite, SCHEMA, SHORT_TABLE_NAME));
+            
+                addIndexIfNotExists(ignite, SCHEMA, SHORT_TABLE_NAME);
+            }
+        }
+        
+        assertFalse(addIndesFut.isDone());
+        assertFalse(addIndesIfNotExistsFut.isDone());
+    
+        ignite1Inhibitor.stopInhibit();
+    
+        assertThrows(IndexAlreadyExistsException.class, () -> {
+            try {
+                addIndesFut.get(10, TimeUnit.SECONDS);
+            } catch (ExecutionException e) {
+                throw e.getCause();
+            }
+        });
+        
+        addIndesIfNotExistsFut.get(10, TimeUnit.SECONDS);
+    }
+    
+    /**
+     * Trys to create a column which is already created.
+     *
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testAddColumn() throws Exception {
+        clusterNodes.forEach(ign -> assertNull(ign.tables().table(TABLE_NAME)));
+        
+        Ignite ignite0 = clusterNodes.get(0);
+        
+        createTable(ignite0, SCHEMA, SHORT_TABLE_NAME);
+        
+        addColumn(ignite0, SCHEMA, SHORT_TABLE_NAME);
```

Comment:
the same - can we remove implicit column name and append it as a param ?

Reply:
Do you see an issue in delete column implementation?
It just a demonstration.